src/pages/Itemdetail/Itemdetail.js

```diff
@@ -1,7 +1,272 @@
+import { TbHeading, TbTruckDelivery } from 'react-icons/tb';
+import { RiErrorWarningLine } from 'react-icons/ri';
+import { AiOutlineCheckCircle, AiOutlineCheck } from 'react-icons/ai';
+import { HiHome, HiOutlineArrowNarrowRight } from 'react-icons/hi';
+import { FaRegHeart } from 'react-icons/fa';
 import './Itemdetail.scss';
 
 const Itemdetail = () => {
-  return <></>;
+  const navigate = useNavigate();
+  const [productDetail, setProductDetail] = useState([]);
+  const [productSize, setProductSize] = useState('');
+  const [readmore, setReadmore] = useState(false);
+  const [buttonToggle, setButtonToggle] = useState('');
+  const { id } = useParams;
+  const priceToString = price => {
+    return price.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+  };
+
+  const toggleMenu = () => {
+    setReadmore(readmore => !readmore);
+  };
+
+  const toggleActive = e => {
+    setButtonToggle(prev => {
+      return e.target.value;
+    });
+  };
+
+  const saveSize = event => {
+    setProductSize(event.target.value);
+  };
+
+  const tokenAuthorization = localStorage.getItem('token');
+
+  useEffect(() => {
+    fetch(`http://10.58.52.160:3000/products/${id}`)
+      .then(data => data.json())
+      .then(data => setProductDetail(data.data));
+  }, [id]);
+
+  const sendtoCart = () => {
+    fetch('http://10.58.52.114:3000/carts', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        authorization: tokenAuthorization,
+      },
+      body: JSON.stringify({
+        productId: productDetail[0].id,
+        sizeId: productSize,
+      }),
+    })
+      .then(data => data.json())
+      .then(data => {
+        if (data.MESSAGE === 'SUCCESS') {
+          localStorage.setItem('access-token', data.access_token);
+          alert('Added to shopping cart!');
+        } else if (data.MESSAGE === 'Out of stock') {
+          alert('Sorry, this product is out of stock');
+        } else if (data.MESSAGE === 'KEY_ERROR') {
+          alert('Key-error occured');
+        }
+      });
+  };
+
+  const sendtoWishlist = () => {
+    fetch('http://10.58.52.114:3000/wishlists', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        authorization: tokenAuthorization,
+      },
+      body: JSON.stringify({
+        productId: productDetail[0].id,
+      }),
+    });
+  };
+
+  return (
+    <div className="detailpage">
+      <div className="detailpageDetailSection">
+        <div className="imageList">
+          <img
+            className="productThumbnail"
+            src={productDetail.thumbnailUrl}
+            alt="mainimage"
+          />
+          {readmore
+            ? productDetail.images.map(el => (
+                <img
+                  className="additionalThumbnail"
+                  key={el}
+                  src={el}
+                  alt="additionalimage"
+                />
+              ))
+            : null}
```

Comment:
```suggestion
          {readmore
            && productDetail.images.map(el => (
                <img
                  className="additionalThumbnail"
                  key={el}
                  src={el}
                  alt="additionalimage"
                />
              ))}
```
단축평가를 사용하면 더 깔끔하게 보입니다!

Reply:
삼항 연산자 쓸 생각밖에 못했는데 감사합니다~!